openwisp_utils/admin.py

```diff
@@ -58,3 +60,64 @@ def has_changed(self):
         if self.instance._state.adding:
             return True
         return super().has_changed()
+
+
+class UUIDAdmin(ModelAdmin):
+    """
+    Defines a field name uuid whose value is that
+    of the id of the object
+    """
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.fields = ['uuid']
+        self.readonly_fields = ['uuid']
+
+    def uuid(self, obj):
+        return obj.pk
+
+    def _get_fields(self, fields, request, obj=None):
+        """
+        removes readonly_fields in add view
+        """
+        if obj:
+            return fields
+        new_fields = fields[:]
+        for field in self.readonly_fields:
+            if field in new_fields:
+                new_fields.remove(field)
+        return new_fields
+
+    def get_fields(self, request, obj=None):
+        return self._get_fields(self.fields, request, obj)
+
+    def get_readonly_fields(self, request, obj=None):
+        return self._get_fields(self.readonly_fields, request, obj)
+
+    class Media:
+        js = ('admin/js/jquery.init.js', 'openwisp-utils/js/uuid.js')
+
+    uuid.short_description = _('UUID')
+
+
+class GetUrlAdmin(ModelAdmin):
```

Comment:
I am also not sure if we need `GetUrlAdmin` as discussed [here](https://github.com/openwisp/openwisp-utils/issues/40#issuecomment-486012776). If we keep it, we have to come up with a test for it. Can someone help me to come up with the idea for the test then?